avaje-config/src/main/java/io/avaje/config/URILoaders.java

```diff
@@ -0,0 +1,40 @@
+package io.avaje.config;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+/** Holds the non-properties ConfigParsers. */
+public final class URILoaders {
+
+  private final Map<String, URIConfigLoader> parserMap = new HashMap<>();
+
+  URILoaders(List<URIConfigLoader> loaders) {
+    for (var parser : loaders) {
+
+      parserMap.put(parser.supportedScheme(), parser);
+    }
+  }
+
+  /** Return the extension ConfigParser pairs. */
+  public Set<Entry<String, URIConfigLoader>> entrySet() {
+    return parserMap.entrySet();
+  }
+
+  /** Return the ConfigParser for the given extension. */
+  public URIConfigLoader get(String extension) {
```

Comment:
Like the ConfigParser this is also `@Nullable` and needs either javadoc or name.

Reply:
later